src/main/scala/mesosphere/marathon/core/instance/Instance.scala

```diff
@@ -77,6 +77,16 @@ case class Instance(
 
   override def region: Option[String] = agentInfo.flatMap(_.region)
 
+  /**
+    * Factory method for creating a scheduled instance from this instance.
+    *
+    * @return new instance with same id in scheduled state.
+    */
+  def rescheduled(): Instance = {
+    val state = InstanceState(Condition.Scheduled, Timestamp.now(), None, None, Goal.Running)
+    Instance(instanceId, None, state, Map.empty, version, unreachableStrategy, None)
```

Comment:
I also don't know if this will work. Given instance A and instance A', if A actually isn't terminal but comes back later, how do we know that the updates for A don't apply to the replacement A' ? Also, Mesos will reject a launch attempt for A if it knows A' to be active.